nncf/onnx/graph/onnx_graph.py

```diff
@@ -30,20 +30,14 @@ class ONNXGraph:
 
     def __init__(self, onnx_model: onnx.ModelProto):
         self.onnx_model = onnx_model
-        self._node_name_to_node = None  # type: Dict[str, onnx.onnx.NodeProto]
-        self._activations_tensor_name_to_value_info = None  # type: Dict[str, onnx.onnx.ValueInfoProto]
-
-    def _update_activation_tensors(self, do_shape_inference: bool = False) -> None:
-        if do_shape_inference:
-            self.onnx_model = onnx.shape_inference.infer_shapes(self.onnx_model)
-        self._activations_tensor_name_to_value_info = {tensor.name: tensor for tensor in
-                                                       self.onnx_model.graph.value_info}
-        model_inputs_name_to_value_info = {tensor.name: tensor for tensor in self.onnx_model.graph.input}
-        model_outputs_name_to_value_info = {tensor.name: tensor for tensor in self.onnx_model.graph.output}
-        initializers = {tensor.name: tensor for tensor in self.onnx_model.graph.initializer}
-        self._activations_tensor_name_to_value_info.update(model_inputs_name_to_value_info)
-        self._activations_tensor_name_to_value_info.update(model_outputs_name_to_value_info)
-        self._activations_tensor_name_to_value_info.update(initializers)
+        self._node_name_to_node = None  # type: Dict[str, onnx.NodeProto]
+        self._edge_name_to_value_info = None  # type: Dict[str, onnx.ValueInfoProto]
+
+    def _update_edges(self) -> None:
+        self.onnx_model = onnx.shape_inference.infer_shapes(self.onnx_model)
```

Comment:
I updated this logic. Now shape inference is done only once using a lazy initialization approach.
It will demand to create of a new ONNXGraph if the model was updated.